map.go

```diff
@@ -202,7 +214,7 @@ func newMapWithOptions(spec *MapSpec, opts MapOptions, btfs btfHandleCache) (_ *
 		defer closeOnError(m)
 
 		if err := spec.checkCompatibility(m); err != nil {
-			return nil, fmt.Errorf("use pinned map %s: %s", spec.Name, err)
+			return nil, fmt.Errorf("%w: %s", NewErrMapIncompatible(spec.Name), err)
```

Comment:
This doesn't work because `%w` has to be at the end of fmt.Errorf if I remember correctly. My suggestion, stick to the simple exported error `var ErrMapIncompatible = ...`.

```suggestion
			return nil, fmt.Errorf("use pinned map %s: %w", spec.Name, err)
```

In checkCompatibility you can then add additional information using the same trick `return fmt.Errorf("details of what is wrong %d: %w", ErrMapIncompatible)`.